mitmproxy/net/udp.py

```diff
@@ -77,11 +80,38 @@ async def wait_closed(self) -> None:
         await self._closed.wait()
 
 
+class WireGuardDatagramTransport(asyncio.DatagramTransport):
+    def __init__(self, server: wg.Server, local_addr: Address, remote_addr: Address):
+        self._server: wg.Server = server
+        self._local_addr: Address = local_addr
+        self._remote_addr: Address = remote_addr
+        self._protocol: DrainableDatagramProtocol | None = None
+        super().__init__()
+
+    def sendto(self, data, addr=None):
+        self._server.send_datagram(data, self._local_addr, addr or self._remote_addr)
+
+    def get_extra_info(self, name: str, default: Any = None) -> Any:
+        if name == "sockname":
+            return self._server.getsockname()
+        else:
+            raise NotImplementedError
+
+    def get_protocol(self):
+        return self
```

Comment:
Similar problem here. `self` is a `Transport` instance, not a `Protocol` instance.